### PR TITLE
End subscription when customer.subscription.deleted

### PR DESCRIPTION
**Pull request by PopDaph**
On the Stripe webhook, we end the subscription when we receive the "customer.subscription.deleted" event.

### CODE REVIEW

File: front/pages/api/stripe/webhook.ts [hunk 1]
```diff
@@ -171,6 +171,36 @@ async function handler(
           // The stripe subscription becomes "past_due".
           // We keep active and email the user and us to manually manage those cases first?
           break;
+        case "customer.subscription.updated":
```
Comment (fontanierh) [inline front/pages/api/stripe/webhook.ts:174]: If we don't do anything, why is it not treated like other events we don't handle ? Ie do nothing as part of `default:` ?

File: front/pages/api/stripe/webhook.ts [hunk 2]
```diff
@@ -171,6 +171,36 @@ async function handler(
...
+          // Occurs when the subscription is updated:
+          // - when the number of seats changes for a metered billing.
+          // - when the subscription is canceled by the user: it is ended at the of the billing period, and we will receive a "customer.subscription.deleted" event.
+          // - when the subscription is activated again after being canceled but before the end of the billing period.
+          break;
+        case "customer.subscription.deleted":
+          // Occurs when the subscription is canceled by the user or by us.
+          const stripeSubscription = event.data.object as Stripe.Subscription;
+          if (stripeSubscription.status === "canceled") {
```
Comment (fontanierh) [inline front/pages/api/stripe/webhook.ts:183]: FMI, in which case would we receive this webhook but not have the cancelled status ?
Also, I always thought it was `cancelled`  but looks like americans use `canceled` 😳  TIL

**Review by fontanierh — DISMISSED** [review]
LGTM, just a small question

File: front/pages/api/stripe/webhook.ts (hunk 2 above)
Comment (fontanierh) [reply front/pages/api/stripe/webhook.ts:183]: If we get the webhook but not the status, should it be an error ?

Comment (PopDaph) [reply front/pages/api/stripe/webhook.ts:183]: Actually they are also unsure about the single or two 'l' 😄 , this is the content of the event: 
Capture d’écran 2023-11-02 à 11 48 31

File: front/pages/api/stripe/webhook.ts (hunk 1 above)
Comment (PopDaph) [reply front/pages/api/stripe/webhook.ts:182]: I added some loggers. To start with, I'd like to be notified when the event occurs to make sure there is not a big hole in the racket

**Review by fontanierh — APPROVED** [review]

**Review by spolu — APPROVED** [review]
:+1:

**Pull request merged**